app/src/main/java/io/gnosis/safe/ui/transaction/list/TransactionPagingSource.kt

```diff
@@ -0,0 +1,40 @@
+package io.gnosis.safe.ui.transaction.list
+
+import androidx.paging.PagingSource
+import io.gnosis.data.models.SafeInfo
+import io.gnosis.data.models.Transaction
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.data.repositories.TransactionRepository
+import pm.gnosis.model.Solidity
+import retrofit2.HttpException
+import java.io.IOException
+
+
+class TransactionPagingSource(
+    private val safe: Solidity.Address,
+    private val safeRepo: SafeRepository,
+    private val txRepo: TransactionRepository
+) : PagingSource<String, Transaction>() {
+
+    private lateinit var safeInfo: SafeInfo
```

Comment:
Would it make sense to pass this as a constructor parameter? Then you can just require it upon construction of this object.

Reply:
I would keep it this way. Right now loading of tx pages is completely encapsulated in the source. Having safeInfo as a parameter would mean we have to load it outside and pass it down to the pager. This increase view model code clutter

Reply:
Well, at the viewModel you have a coroutine scope from which you can launch anyway, then this class states in its constructor all the parameters it needs instead of requiring stuff that may not be initialised when it's needed.